keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt

```diff
@@ -29,9 +42,13 @@ class EnvironmentPromotionChecker(
               versions.firstOrNull()
             } else {
               versions.firstOrNull { v ->
-                constraints.all { constraintEvaluator ->
-                  !environment.hasSupportedConstraint(constraintEvaluator) || constraintEvaluator.canPromote(artifact, v, deliveryConfig, environment.name)
-                }
+                /**
+                 * Only check stateful evaluators if all stateless evaluators pass. We don't
```

Comment:
This actually brings up a question on the ordering of constraints for me. Since you can't tell a `List` from `Set` in YAML, I suspect users might assume order matters when specifying constraints for an environment (in which case it would not be OK to make this assumption here), even though they're modeled as a `Set` internally. Should we perhaps clarify that order does _not_ matter in the documentation?

Reply:
Sure, please do update documentation :)

Reply:
Is the `allowed-times` constraint rolled out to main yet, BTW? I can add an example of that while I'm at it.